### PR TITLE
[VEN-1289]: Swap Audit Fixes

### PR DESCRIPTION
**Pull request by defcon022**
## Description

This PR contains fixes suggested during the Swap audit:
- PVE001: Revisited slippage control for supporting fee tokens.
- PVE002: Improved sanity checks for system parameters.
- Naming consistency in swap router.
- Improved gas efficiency in router helper.

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

**Review by chechu — APPROVED** [review]

**Review by GitGuru7 — APPROVED** [review]

File: contracts/Swap/interfaces/IPancakeSwapV2Router.sol [hunk 1]
```diff
@@ -93,7 +93,7 @@ interface IPancakeSwapV2Router {
         uint256 deadline
     ) external payable;
 
-    function swapBnbAndSupplyAtSupportingFee(
+    function swapBNBAndSupplyAtSupportingFee(
```
Comment (therealemjy) [inline contracts/Swap/interfaces/IPancakeSwapV2Router.sol:96]: To follow the new naming convention I'd suggest:
```suggestion
    function swapExactBNBForTokensAndSupplyAtSupportingFee(
```

**Review by therealemjy — COMMENTED** [review]
I'd suggest renaming a few more functions to follow the new naming convention:

`swapAndSupply` -> `swapExactTokensForTokensAndSupply`
`swapAndSupplyAtSupportingFee ` -> `swapExactTokensForTokensAndSupplyAtSupportingFee`
`swapBnbAndSupply ` -> `swapExactBNBForTokensAndSupply`

File: contracts/Swap/interfaces/IPancakeSwapV2Router.sol [hunk 2]
```diff
@@ -131,14 +131,14 @@ interface IPancakeSwapV2Router {
         uint256 deadline
     ) external;
 
-    function swapBnbAndRepay(
+    function swapBNBAndRepay(
```
[the remark this answers is not in the document]
Comment (defcon022) [reply contracts/Swap/interfaces/IPancakeSwapV2Router.sol:134]: Fixed.

File: contracts/Swap/interfaces/IPancakeSwapV2Router.sol [hunk 3]
```diff
@@ -131,14 +131,14 @@ interface IPancakeSwapV2Router {
...
         address vTokenAddress,
         uint256 amountOutMin,
         address[] calldata path,
         uint256 deadline
     ) external payable;
 
-    function swapBnbAndRepayAtSupportingFee(
+    function swapBNBAndRepayAtSupportingFee(
```
[the remark this answers is not in the document]
Comment (defcon022) [reply contracts/Swap/interfaces/IPancakeSwapV2Router.sol:141]: Fixed.

File: contracts/Swap/interfaces/IPancakeSwapV2Router.sol (hunk 1 above)
Comment (defcon022) [reply contracts/Swap/interfaces/IPancakeSwapV2Router.sol:96]: Fixed.

File: contracts/Swap/SwapRouter.sol [hunk 4]
```diff
@@ -592,11 +592,11 @@ contract SwapRouter is Ownable2StepUpgradeable, RouterHelper, IPancakeSwapV2Rout
         address[] calldata path,
         address to,
         uint256 deadline
-    ) external override ensure(deadline) ensurePath(path) returns (uint256[] memory amounts) {
+    ) external override ensure(deadline) ensurePath(path) returns (uint256 swapAmount) {
```
Comment (chechu) [inline contracts/Swap/SwapRouter.sol:595]: Why did you change the returned value? out of curiosity... because now it seems incoherent with `swapExactTokensForBNB`, for example

File: contracts/Swap/interfaces/IPancakeSwapV2Router.sol (hunk 1 above)
Comment (defcon022) [reply contracts/Swap/interfaces/IPancakeSwapV2Router.sol]: My bad, pushed it now. :-)

File: contracts/Swap/SwapRouter.sol (hunk 4 above)
Comment (defcon022) [reply contracts/Swap/SwapRouter.sol:595]: Because the swap for tokens at a supporting fee will only return the swapped amount instead of an array of all amounts(swapped between) as the output of these amounts can not be assumed in the start of the Swap.

**Review by GitGuru7 — APPROVED** [review]
